raiden_contracts/tests/test_channel_settle.py

```diff
@@ -193,6 +197,244 @@ def equivalent_transfers(balance_proof):
         )
 
 
+def test_settle_single_direct_transfer_for_closing_party(
+        web3,
+        get_accounts,
+        token_network,
+        create_channel,
+        channel_deposit,
+        create_balance_proof,
+):
+    """ Test settle of a channel with one direct transfer to the participant
+    that called close.
+    """
+    (A, B) = get_accounts(2)
+    (vals_A, vals_B) = (
+        ChannelValues(deposit=1, withdrawn=0, transferred=0, locked=0),
+        ChannelValues(deposit=10, withdrawn=0, transferred=5, locked=0),
+    )
+    settle_timeout = TEST_SETTLE_TIMEOUT_MIN
+    locksroot = fake_bytes(32)
+
+    channel_identifier = create_channel(A, B)[0]
+    channel_deposit(A, vals_A.deposit, B)
+    channel_deposit(B, vals_B.deposit, A)
+
+    balance_proof_B = create_balance_proof(
+        channel_identifier,
+        B,
+        vals_B.transferred,
+        vals_B.locked,
+        1,
+        locksroot,
+    )
+    token_network.functions.closeChannel(B, *balance_proof_B).transact({'from': A})
+
+    web3.testing.mine(settle_timeout)
+    token_network.functions.settleChannel(
+        A,
+        0,
+        0,
+        locksroot,
+        B,
+        vals_B.transferred,
+        0,
+        locksroot,
+    ).transact({'from': A})
+
+    # Calculate how much A and B should receive
+    expected_settlement = get_settlement_amounts(vals_A, vals_B)
+    # Calculate how much A and B receive according to onchain computation
+    onchain_settlement = get_onchain_settlement_amounts(vals_A, vals_B)
+
+    assert (expected_settlement.participant1_balance == onchain_settlement.participant1_balance)
+    assert (expected_settlement.participant2_balance == onchain_settlement.participant2_balance)
```

Comment:
We need checks on the actual on-chain balances of the participants, as it was in the original `raiden` test:
This needs to be reproduced:
```
assert tester_token.balanceOf(address0, sender=pkey0) == initial0 + deposit - amount
assert tester_token.balanceOf(address1, sender=pkey0) == initial1 + deposit + amount
assert tester_token.balanceOf(nettingchannel.address, sender=pkey0) == 0
```

Reply:
Made a PR for this: https://github.com/raiden-network/raiden-contracts/pull/174